extensions/amp-story/1.0/test/test-amp-story-interactive-results.js

```diff
@@ -87,12 +105,108 @@ describes.realWin(
+        'results-threshold',
+      ]);
+      expect(decideStrategy(ampStoryResults.parseOptions_())).to.equal(
+        'percentage'
+      );
+    });
+
+    it('should use a category strategy if there are no thresholds', () => {
+      addConfigToInteractive(ampStoryResults, 3, null, ['results-category']);
+      expect(decideStrategy(ampStoryResults.parseOptions_())).to.equal(
+        'category'
+      );
+    });
+
+    it('should select the category with most votes', () => {
+      const votes = {
+        'a': {
+          'option': {'resultscategory': 'results-category 1'},
+          'type': InteractiveType.POLL,
+        },
+        'b': {
+          'option': {'resultscategory': 'results-category 2'},
+          'type': InteractiveType.POLL,
+        },
+        'c': {
+          'option': {'resultscategory': 'results-category 2'},
+          'type': InteractiveType.POLL,
+        },
+      };
+      addConfigToInteractive(ampStoryResults, 3, null, ['results-category']);
+      expect(
+        processResultsCategory(votes, ampStoryResults.parseOptions_()).category
+      ).to.equal('results-category 2');
+    });
+
+    it('should select the first category with most votes if tied', () => {
+      const votes = {
+        'a': {
+          'option': {'resultscategory': 'results-category 1'},
+          'type': InteractiveType.POLL,
+        },
+        'b': {
+          'option': {'resultscategory': 'results-category 2'},
+          'type': InteractiveType.POLL,
+        },
+      };
+      addConfigToInteractive(ampStoryResults, 3, null, ['results-category']);
+      expect(
+        processResultsCategory(votes, ampStoryResults.parseOptions_()).category
+      ).to.equal('results-category 1');
+    });
+
+    it('should get the highest threshold that is lower or equal to the percentage', () => {
+      const votes = {
+        'a': {
+          'option': {'correct': 'correct'},
+          'type': InteractiveType.QUIZ,
+        },
+        'b': {
+          'option': {},
+          'type': InteractiveType.QUIZ,
+        },
+      };
+      addThresholdsToInteractive(ampStoryResults, [25, 80, 50]);
+      const results = processResultsPercentage(
+        votes,
+        ampStoryResults.parseOptions_()
+      );
+      expect(results.category).to.equal('results-category 3');
+      expect(results.percentage).to.equal(50);
+    });
+
+    it('should get the lowest threshold if all are higher than percentage', () => {
+      const votes = {
+        'a': {
+          'option': {},
+          'type': InteractiveType.QUIZ,
+        },
+        'b': {
+          'option': {},
+          'type': InteractiveType.QUIZ,
+        },
+      };
+      addThresholdsToInteractive(ampStoryResults, [50, 20, 80]);
+      const results = processResultsPercentage(
+        votes,
+        ampStoryResults.parseOptions_()
+      );
+      expect(results.category).to.equal('results-category 2');
+      expect(results.percentage).to.equal(0);
+    });
```

Comment:
Can you either update these tests so they're more "end to end" like the one line 101, or add new ones?

Reply:
Done, added 2 tests to check the title of the category is as expected, and the percentage number (top right) is as expected